great_expectations/execution_engine/sqlalchemy_execution_engine.py

```diff
@@ -1149,12 +1149,8 @@ def execute_split_query(self, split_query: Selectable) -> List[Row]:
             pattern = re.compile(r"(CAST\(EXTRACT\(.*?\))( AS STRING\))", re.IGNORECASE)
             split_query = re.sub(pattern, r"\1 AS VARCHAR)", split_query)
 
-        if isinstance(self.engine, sqlalchemy_Engine):
-            connection = self.engine.connect()
-        else:
-            connection = self.engine
-
-        with connection:
+        # ensuring that we are executing the split query with an sa.Engine and not sa.Connection with engine.engine.
+        with self.engine.engine.begin() as connection:
```

Comment:
This had to be a guaranteed to be an `sa.Engine`, or else we would close the Connection

Reply:
Can we use something more like this pattern here? (copied from map_condition_auxiliary_methods.py) or will that not work?
```
if sqlalchemy_Engine and isinstance(execution_engine.engine, sqlalchemy_Engine):
            connection = execution_engine.engine.connect()
        else:
            # execution_engine.engine is already a Connection. Use it directly
            connection = execution_engine.engine
```

Reply:
I'm not 100% sure, so I'm submitting one more time with the proposed changes. If things break, then I'll revert back to this `begin()` pattern.